src/AutoMapper/TypeDetails.cs

```diff
@@ -107,14 +107,17 @@ private IEnumerable<MethodInfo> BuildPublicNoArgExtensionMethods(IEnumerable<Met
             {
                 genericInterfaces = genericInterfaces.Union(new[] { Type });
             }
+
             return explicitExtensionMethods.Union
             (
-                from genericMethod in sourceExtensionMethodSearch
-                where genericMethod.IsGenericMethodDefinition
                 from genericInterface in genericInterfaces
                 let genericInterfaceArguments = genericInterface.GetTypeInfo().GenericTypeArguments
-                where genericMethod.GetGenericArguments().Length == genericInterfaceArguments.Length
-                let methodMatch = genericMethod.MakeGenericMethod(genericInterfaceArguments)
+                from extensionMethod in sourceExtensionMethodSearch
+                where !extensionMethod.IsGenericMethodDefinition
+                    || extensionMethod.GetGenericArguments().Length == genericInterfaceArguments.Length
+                let methodMatch = extensionMethod.IsGenericMethodDefinition
+                    ? extensionMethod.MakeGenericMethod(genericInterfaceArguments)
+                    : extensionMethod
```

Comment:
I think two queries linked by Concat make more sense here. One for generic methods and one for the others.